dm/syncer/safe_mode.go

```diff
@@ -43,35 +53,68 @@ func (s *Syncer) enableSafeModeInitializationPhase(tctx *tcontext.Context) {
 		s.safeMode.Add(tctx, 1) // add 1 but has no corresponding -1, so keeps enabled
 		s.tctx.L().Info("enable safe-mode by config")
 	}
-	if s.checkpoint.SafeModeExitPoint() != nil {
+
+	var duration time.Duration
```

Comment:
the duration has complicated read and write, can we change the logic to

if exitPoint != nil {
  check checkpoint is equal to exitPoint, return nil
  check has "0s" safe-mode-duration, return error
} else {
  turn on safe mode for safe-mode-duration
}

Reply:
yes, you are the right. I'm sry I forgot it😂